include/exadg/fluid_structure_interaction/precice/exadg_coupling.h

```diff
@@ -38,13 +38,16 @@ template<int dim, int data_dim, typename VectorizedArrayType>
 class ExaDGCoupling : public CouplingBase<dim, data_dim, VectorizedArrayType>
 {
 public:
+  static unsigned int const rank =
+    (data_dim == 1) ? 0 : ((data_dim == dim) ? 1 : dealii::numbers::invalid_unsigned_int);
+
   ExaDGCoupling(
     std::shared_ptr<dealii::MatrixFree<dim, double, VectorizedArrayType> const> data,
 #ifdef EXADG_WITH_PRECICE
     std::shared_ptr<precice::SolverInterface> precice,
 #endif
-    std::string const                                              mesh_name,
-    std::shared_ptr<ContainerInterfaceData<dim, data_dim, double>> interface_data_,
```

Comment:
Question: what is the reason for removing `double`?

Reply:
Types other than `double` are not supported for `FunctionCached`, so I think it is best to remove the template argument. Of course, I could still introduce it for `ContainerInterfaceData` but only use it with double. Should I do this?

Reply:
see https://github.com/exadg/exadg/pull/230/commits/f6289027e86ef9ac1aaa4b9c4c2dd06e77821711